CRM/Financial/BAO/FinancialType.php

```diff
@@ -243,18 +243,20 @@ public static function getAllEnabledAvailableFinancialTypes($action = CRM_Core_A
    *
    * @return array
    */
-  public static function getAvailableFinancialTypes(&$financialTypes = NULL, $action = CRM_Core_Action::VIEW, $unused = FALSE, $includeDisabled = FALSE) {
```

Comment:
If the reference is not actually used for anything, we should just remove the `&`.

Reply:
Unfortunately it is, it's used in the test at the very least.

Reply:
@MegaphoneJon looking at the test fails it seems to be pointing to a situation here where somehow $financialTypes is returning a NULL value e.g. we are getting in some of the test fails a type error on this line https://github.com/civicrm/civicrm-core/blob/master/CRM/Price/BAO/PriceFieldValue.php#L88 because $financialTypes is NULL

Reply:
Thanks for picking that up @seamuslee001.  The issue is managing both the return-by-reference and a cached value when the function is called multiple times. I think I've got it now though.

Reply:
@MegaphoneJon I found one other issue in the way the early return for if not in financialTypeACLs was being returned in regards to the passbyReference which I have just pushed a commit to fix which I think might fix a bunch of the unit test failures